### ISSUE TITLE
Add option to not urlencode links to files with special characters in them

### ISSUE THREAD

**Issue by natemccurdy** [buunguyen/octotree#274]
This isn't really a bug, but maybe more of a feature request or change in default behavior.

When using octotree to browse repository files, if a file has a special character in its name or path, like a `-`, `/`, or ` `, the link's URL has those characters converted to the `%#` equivalent.

For example, in this repo, this:
```
https://github.com/buunguyen/octotree/blob/master/libs/jstree.css
```
ends up as:
```
https://github.com/buunguyen/octotree/blob/master/libs%2Fjstree.js
```

I think that not changing those characters would result in cleaner URL's when copy/pasting from the Octotree tree view, and would also result in having the visited URL match what a user would visit had they not used Octotree to get there.

**Comment by elifoster** [comment]
I feel like this is can definitely be considered a bug, since none of those characters should be considered "special" in a URL.

**Comment by buunguyen** [comment]
They're considered special by `encodeURIComponents`. [This line](https://github.com/buunguyen/octotree/blob/master/src/adapters/adapter.js#L77) encodes the path. Removing this will break repo with paths containing stuff like "?", "#" etc.

Maybe a solution is splitting the path by '/' first, then encode individual components before merging them. But I need to make sure there's no edge case that can break this.

**Comment by wuliupo** [comment]
![screenshot](https://cloud.githubusercontent.com/assets/1187809/15310404/178b3fa2-1c22-11e6-980e-ae5b25730e87.png)

**Comment by buunguyen** [comment]
Fixed in 2.0.10.
